### PR TITLE
Workbench linux install

### PR DESCRIPTION
**Pull request by peterfpeterson**
The purpose of this work is to add the workbench to the nightly installer for rhel7, but changes to the ubuntu installer were also done. Since `buildscript` already enables the workbench on PRs and master, that script does not need to be altered.

Small cleanup in the other linux launch scripts happened as well. Notably, `launch_mantidplot.sh` is now installed as `mantidplot` so there is one less symbolic link created.

**To test:**

Run the rhel7 installer (build and run for ubuntu as well) somewhere and see that the following all "work" without errors or warnings.
* `/opt/mantidunstable/bin/mantidplot`
* `/opt/mantidunstable/bin/MantidPlot`
* `/opt/mantidunstable/bin/mantidpython`
* `/opt/mantidunstable/bin/mantidworkbench`

Bonus items for RHEL7
* `/usr/bin/mantidplotunstable`
* `/usr/bin/mantidpythonunstable`
* `/usr/bin/mantidworkbenchunstable`

**Tested supported platforms:**
* RHEL7 *github username*
* Ubuntu 16.04 @martyngigg
* Ubuntu 18.04 @AndreiSavici 
* Windows (workbench not in installer, but rest should still work) @martyngigg 
* MacOS (workbench not in installer, but rest should still work) @quantumsteve 

*There is no associated issue.*

*This should not be in the release notes yet* because it is a step in the direction of having something for users to try and comment on.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: qt/python/mantidqt/CMakeLists.txt [hunk 1]
```diff
@@ -18,6 +17,13 @@ list ( APPEND common_link_libs
   ${PYTHON_LIBRARIES}
 )
 
+# TODO - Should the macro below do this? There is only one ui
+#        file shared between qt4/qt5
+set (UI_FILES dialogs/spectraselectordialog.ui)
```
Comment (martyngigg) [inline qt/python/mantidqt/CMakeLists.txt]: I think we can do this as part of the `setup.py` to gather extra files.

**Comment by AndreiSavici** [comment]
Tried it on Ubuntu 18.04. 
When I try to run mantidplot, I get 
```
DllOpen-[Error] Could not open library /opt/mantidunstable/plugins/libMantidVatesAlgorithms.so: libvtkIOLegacy-pv5.4.so.1: cannot open shared object file: No such file or directory
DllOpen-[Error] Could not open library /opt/mantidunstable/plugins/libMantidVatesAlgorithms.so: libvtkIOLegacy-pv5.4.so.1: cannot open shared object file: No such file or directory
```
The file is in `/opt/mantidunstable/lib/paraview-5.4`
When I try to run the workbench, I get 
```
Traceback (most recent call last):
  File "/opt/mantidunstable/lib/mantidqt/utils/qt/__init__.py", line 57, in import_qt
    lib = import_module(modulename + LIB_SUFFIX, package)
  File "/usr/lib/python3.6/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 994, in _gcd_import
  File "<frozen importlib._bootstrap>", line 971, in _find_and_load
  File "<frozen importlib._bootstrap>", line 953, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'mantidqt.widgets._commonqt5'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/mantidunstable/lib/workbench/app/mainwindow.py", line 428, in main
    main_window = start_workbench(app)
  File "/opt/mantidunstable/lib/workbench/app/mainwindow.py", line 400, in start_workbench
    main_window.setup()
  File "/opt/mantidunstable/lib/workbench/app/mainwindow.py", line 185, in setup
    from workbench.plugins.workspacewidget import WorkspaceWidget
  File "/opt/mantidunstable/lib/workbench/plugins/workspacewidget.py", line 26, in <module>
    from mantidqt.widgets.workspacewidget.workspacetreewidget import WorkspaceTreeWidget
  File "/opt/mantidunstable/lib/mantidqt/widgets/workspacewidget/workspacetreewidget.py", line 28, in <module>
    'WorkspaceTreeWidgetSimple')
  File "/opt/mantidunstable/lib/mantidqt/utils/qt/__init__.py", line 59, in import_qt
    lib = import_module(modulename.lstrip('.') + LIB_SUFFIX)
  File "/usr/lib/python3.6/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
RuntimeError: _commonqt5 cannot import type '����' from PyQt5.QtCore

********* UNHANDLED EXCEPTION *********
  what(): System exception

Backtrace:
 /opt/mantidunstable/lib/libMantidAPI.so(+0x199159) [0x7f3e2fd36159]
 /opt/mantidunstable/lib/libMantidAPI.so(+0x199300) [0x7f3e2fd36300]
 /usr/lib/x86_64-linux-gnu/libstdc++.so.6(+0x92d3a) [0x7f3e58bf4d3a]
 /usr/lib/x86_64-linux-gnu/libstdc++.so.6(+0x91d59) [0x7f3e58bf3d59]
 /usr/lib/x86_64-linux-gnu/libstdc++.so.6(__gxx_personality_v0+0x2e8) [0x7f3e58bf4708]
 /lib/x86_64-linux-gnu/libgcc_s.so.1(+0x10763) [0x7f3e5895a763]
 /lib/x86_64-linux-gnu/libgcc_s.so.1(_Unwind_RaiseException+0x2b1) [0x7f3e5895acd1]
 /usr/lib/x86_64-linux-gnu/libstdc++.so.6(__cxa_throw+0x3b) [0x7f3e58bf4fdb]
 /usr/lib/libPocoFoundation.so.50(_ZN4Poco10MemoryPool7releaseEPv+0xbf) [0x7f3e2f3cb95f]
 /usr/lib/libPocoNet.so.50(_ZN4Poco3Net19HTTPHeaderStreamBufD1Ev+0x1f) [0x7f3e2f053e9f]
 /usr/lib/libPocoNet.so.50(_ZN4Poco3Net21HTTPHeaderInputStreamD1Ev+0x40) [0x7f3e2f054280]
 /usr/lib/libPocoNet.so.50(_ZN4Poco3Net17HTTPClientSession15receiveResponseERNS0_12HTTPResponseE+0xe8) [0x7f3e2f04d528]
 /opt/mantidunstable/lib/../lib/libMantidKernel.so(_ZN6Mantid6Kernel15GitHubApiHelper21sendRequestAndProcessERN4Poco3Net17HTTPClientSessionERNS2_3URIERSo+0x6a) [0x7f3e2e76b66a]
 /opt/mantidunstable/lib/../lib/libMantidKernel.so(_ZN6Mantid6Kernel14InternetHelper16sendHTTPSRequestERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERSo+0x348) [0x7f3e2e77ccb8]
 /opt/mantidunstable/lib/../lib/libMantidKernel.so(_ZN6Mantid6Kernel14InternetHelper11sendRequestERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERSo+0x1a4) [0x7f3e2e7766e4]
 /opt/mantidunstable/lib/../lib/libMantidKernel.so(_ZN6Mantid6Kernel14InternetHelper12downloadFileERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEES9_+0xd3) [0x7f3e2e77d043]
 /opt/mantidunstable/plugins/libMantidDataHandling.so(_ZN6Mantid12DataHandling18DownloadInstrument14doDownloadFileERKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEES9_RKSt3mapIS7_S7_St4lessIS7_ESaISt4pairIS8_S7_EEE+0x95) [0x7f3e20acecd5]
 /opt/mantidunstable/plugins/libMantidDataHandling.so(_ZN6Mantid12DataHandling18DownloadInstrument17processRepositoryB5cxx11Ev+0x5f6) [0x7f3e20ad0326]
 /opt/mantidunstable/plugins/libMantidDataHandling.so(_ZN6Mantid12DataHandling18DownloadInstrument4execEv+0x15f) [0x7f3e20ad182f]
 /opt/mantidunstable/lib/libMantidAPI.so(_ZN6Mantid3API9Algorithm4execENS_8Parallel13ExecutionModeE+0x43) [0x7f3e2fc9a5b3]
 /opt/mantidunstable/lib/libMantidAPI.so(_ZN6Mantid3API9Algorithm7executeEv+0x3c8) [0x7f3e2fca0fd8]
 /opt/mantidunstable/lib/libMantidAPI.so(_ZN6Mantid3API9Algorithm16executeAsyncImplERKN4Poco4VoidE+0x14) [0x7f3e2fc97c54]
 /opt/mantidunstable/lib/libMantidAPI.so(_ZN6Mantid3API14AlgorithmProxy16executeAsyncImplERKN4Poco4VoidE+0x22) [0x7f3e2fcbe592]
 /opt/mantidunstable/lib/libMantidAPI.so(+0x12327b) [0x7f3e2fcc027b]
 /usr/lib/libPocoFoundation.so.50(_ZN4Poco12PooledThread3runEv+0x8f) [0x7f3e2f40d52f]
 /usr/lib/libPocoFoundation.so.50(_ZN4Poco10ThreadImpl13runnableEntryEPv+0x99) [0x7f3e2f408de9]
 /lib/x86_64-linux-gnu/libpthread.so.0(+0x76db) [0x7f3e59edc6db]
 /lib/x86_64-linux-gnu/libc.so.6(clone+0x3f) [0x7f3e5901188f]
QObject::~QObject: Timers cannot be stopped from another thread
Traceback (most recent call last):
  File "/opt/mantidunstable/scripts/ErrorReporter/error_dialog_app.py", line 15, in <module>
    import mantid # noqa
ModuleNotFoundError: No module named 'mantid'
```

File: qt/python/mantidqt/CMakeLists.txt [hunk 2]
```diff
@@ -59,5 +59,7 @@ mtd_add_sip_module (
     ${PYTHON_LIBRARIES}
     ${Boost_LIBRARIES}
     API
+  INSTALL_DIR
+    ${LIB_DIR}/mantidqt
```
Comment (quantumsteve) [inline qt/python/mantidqt/CMakeLists.txt]: This brings in `_commonqt5.so` to the MacOS bundle. Can you add an `if` so that it only installs on Linux?

Comment (martyngigg) [reply qt/python/mantidqt/CMakeLists.txt]: Oh. I thought I had managed to disabled that with https://github.com/mantidproject/mantid/pull/23211/files/800f7df7998fdb8ff80e13fe410d8037a7fc06cf#diff-f36073e58ea48a6c694c553145a7f17aR241. I'll take a look.

Does it do this even if `PACKAGE_WORKBENCH=OFF`?

**Comment by martyngigg** [comment]
I've tested it on Ubuntu 16.04 and everything starts okay with Python 2. Perhaps Andrei's issue is a Python 3 problem? I'll see if I can reproduce it on 16.04.

**Comment by martyngigg** [comment]
A Python 3 build for me on Ubuntu 16.04 also works.

File: qt/python/mantidqt/CMakeLists.txt (hunk 2 above)
Comment (quantumsteve) [reply qt/python/mantidqt/CMakeLists.txt]: `_commonqt5.so` isn't copied when `PACKAGE_WORKBENCH=OFF`

Comment (peterfpeterson) [reply qt/python/mantidqt/CMakeLists.txt]: Correct

**Review by quantumsteve — APPROVED** [review]
No issues on MacOS.

**Comment by martyngigg** [comment]
I will just double check windows again.

**Review by martyngigg — APPROVED** [review]
The windows package  looks good.

**Comment by peterfpeterson** [comment]
The linux package isn't currently working. `mantidworkbench` does, but `mantidpython` and `mantidplot` do not.

**Review by rosswhitfield — CHANGES_REQUESTED** [review]
The `MantidPlot_exe` appears to be missing from the RHEL7 package.
```
$ /opt/mantidunstable/bin/mantidplot
/bin/vglrun: line 257: /opt/mantidunstable/bin/MantidPlot_exe: No such file or directory
FrameworkManager-[Notice] Welcome to Mantid 3.13.20180810.1809
FrameworkManager-[Notice] Please cite: http://dx.doi.org/10.1016/j.nima.2014.07.029 and this release: http://dx.doi.org/10.5286/Software/Mantid
DownloadInstrument-[Notice] All instrument definitions up to date
error-[Error] Continue working.
```
```
$ ls /opt/mantidunstable/bin
ipython_widget     mantidpython                   pvdataserver
mantid             mantidqtpython.so              pvpython
mantidplot         mantid_qt_settings_editor.py   pvrenderserver
mantidplot.py      mantid_qt_settings_editor.pyc  pvserver
mantidplot.pyc     mantid_reset_settings.sh       pymantidplot
mantidplotrc.py    mantidworkbench                workbench
mantidplotrc.pyc   paraview-config
Mantid.properties  pvbatch
```

But `mantidworkbench` works :+1:

**Review by AndreiSavici — APPROVED** [review]
Works on Ubuntu 18.04

**Comment by martyngigg** [comment]
I've also just double-checked the upgrade/remove behaviour on Ubuntu 16.04 now you mentioned it - everything still looks fine there.

**Comment by peterfpeterson** [comment]
The [order of scriptlets](https://fedoraproject.org/wiki/Packaging:Scriptlets#Ordering) for upgrade makes the logic of those exceedingly strange. This is complicated by the [limited number of scriptlets that cmake](https://cmake.org/cmake/help/v3.6/module/CPackRPM.html) allows for doing the work in. Copied from fedora itself, the order of scriptlets that is accessible from cmake (with cmake variable names) is
1. `CPACK_RPM_PRE_INSTALL_SCRIPT_FILE` of new package
1. ` CPACK_RPM_POST_INSTALL_SCRIPT_FILE` of new package
1. `CPACK_RPM_PRE_UNINSTALL_SCRIPT_FILE` of old package - ours is empty
1. (removal of old package)
1. `CPACK_RPM_POST_UNINSTALL_SCRIPT_FILE` of old package

**Review by rosswhitfield — APPROVED** [review]
Now it works on RHEL7

**Pull request merged**